src/meltano/cli/state.py

```diff
@@ -310,12 +310,26 @@ def get_state(ctx: click.Context, project: Project, state_id: str) -> None:
 
 @meltano_state.command(cls=InstrumentedCmd, name="clear")
 @prompt_for_confirmation(prompt="This will clear state for the job. Continue?")
-@click.argument("state-id")
+@click.argument("state-id", required=False)
+@click.option(
+    "--all",
```

Comment:
change:

I'm a bit surprised Ruff is not complaining about this shadowing the builtin function `all`, but we should still change the parameter name:

```suggestion
    ["--all", "clear_all"],
```

Reply:
Ruff did complain [here](https://results.pre-commit.ci/run/github/378997214/1726756744.X8yggwgrQWKXwLLkifo0_w) haha!
Anyways, I changed this.